### PR TITLE
initial implementation of sitemap alternate support

### LINKED ISSUE #47: Sitemap support
So mkdocs does generate a sitemap.xml for all pages but currently the extra language pages generated by this plugin do not get added to it, it seems. 

```xml
<url>
    <loc>http://www.example.com/docs/</loc>
    <xhtml:link rel="alternate" hreflang="de" href="http://www.example.com/de/"/>
  </url>
```

This example Is probably what it should do, not sure if plugins can even change the sitemap generation though.

### PR DESCRIPTION
**Pull request by ultrabug**
given that we provide a sitemap.xml template from theme.custom_dir
this allows to generate alternate links to make the sitemap
support multiple languages

https://developers.google.com/search/docs/advanced/crawling/localized-versions#sitemap

closes #47
closes #54

### CODE REVIEW

**Comment by ultrabug** [comment]
The recipe is to have this:

```
diff --git a/mkdocs.yml b/mkdocs.yml
index 4d12f45..7f128ff 100644
--- a/mkdocs.yml
+++ b/mkdocs.yml
@@ -11,18 +11,9 @@ theme:
     logo: material/file-document-multiple
   palette:
     primary: blue
+  custom_dir: custom_theme/
```

and provide a `custom_theme/sitemap.xml` which looks like this:

```
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
{%- for file in pages -%}
    {% if not file.page.is_link %}
    <url>
         <loc>{% if file.page.canonical_url %}{{ file.page.canonical_url|e }}{% else %}{{ file.page.abs_url|e }}{% endif %}</loc>
         {% if file.page.update_date %}<lastmod>{{file.page.update_date}}</lastmod>{% endif %}
         <changefreq>daily</changefreq>
         {% for locale, i18n_file in file.alternates.items() -%}
         {% if i18n_file %}<xhtml:link rel="alternate" hreflang="{{ locale }}" href="{% if i18n_file.page.canonical_url %}{{ i18n_file.page.canonical_url|e }}{% else %}{{ config.site_url }}{{ i18n_file.url }}{% endif %}"/>{% endif %}
         {% endfor %}
    </url>
    {%- endif -%}
{% endfor %}
</urlset>
```

then the built sitemap.xml looks like:

```
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <url>
         <loc>https://ultrabug.github.io/mkdocs-static-i18n/</loc>
         <lastmod>2022-01-28</lastmod>
         <changefreq>daily</changefreq>
         <xhtml:link rel="alternate" hreflang="en" href="https://ultrabug.github.io/mkdocs-static-i18n/en/"/>
         <xhtml:link rel="alternate" hreflang="fr" href="https://ultrabug.github.io/mkdocs-static-i18n/fr/"/>
         
    </url>
    <url>
         <loc>https://ultrabug.github.io/mkdocs-static-i18n/topic1/named_file/</loc>
         <lastmod>2022-01-28</lastmod>
         <changefreq>daily</changefreq>
         <xhtml:link rel="alternate" hreflang="en" href="https://ultrabug.github.io/mkdocs-static-i18n/en/topic1/named_file/"/>
         <xhtml:link rel="alternate" hreflang="fr" href="https://ultrabug.github.io/mkdocs-static-i18n/fr/topic1/named_file/"/>
         
    </url>
    <url>
         <loc>https://ultrabug.github.io/mkdocs-static-i18n/topic2/</loc>
         <lastmod>2022-01-28</lastmod>
         <changefreq>daily</changefreq>
         <xhtml:link rel="alternate" hreflang="en" href="https://ultrabug.github.io/mkdocs-static-i18n/en/topic2/"/>
         <xhtml:link rel="alternate" hreflang="fr" href="https://ultrabug.github.io/mkdocs-static-i18n/fr/topic2/"/>
         
    </url>
</urlset>
```

**Comment by ultrabug** [comment]
FYI @jrappen @Stanzilla I might have overcome the sitemap situation

Now I wonder if the plugin should ship and modify the sitemap.xml template by itself or just add a documentation on how to do it on the user side...

**Comment by Stanzilla** [comment]
Oh nice, I think it should probably do it itself

**Comment by jrappen** [comment]
Same opinion as Ben. Users choose this plugin to do exactly that.

---

Does this PR also fix [`navigation.instant`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-loading) issues described in #62?

**Comment by ultrabug** [comment]
Alright, I've made it so that the plugin takes care of everything by itself :star:

@Stanzilla @jrappen would you by any chance be able to test this?

**Comment by ultrabug** [comment]
> Does this PR also fix navigation.instant issues described in #62?

I'm not sure since I don't really know the internals of the JS... we'll see!

File: README.md [hunk 1]
```diff
@@ -92,6 +92,22 @@ site
     └── index.html
 ```
 
+### Alternate aware sitemap.xml
+
+The `mkdocs-static-i18n` provides and will generate an alternate aware `sitemap.xml` for you automatically so that your localized content is made available to search engines!
+
+Localization aware sitemap.xml:
+
+```
```
Comment (jrappen) [inline README.md:101]: should be:

````
    ```xml
````

**Comment by Stanzilla** [comment]
Not sure how much time I will have in the near future but I can try!

**Comment by Stanzilla** [comment]
Found the time to test this, it looks like it adds an alternate link to the en/default site lang link as well, which seems wrong?

config:
```yml
  - i18n:
      default_language: !ENV [DEFAULT_LANGUAGE, "en"]
      default_language_only: !ENV [DEFAULT_LANGUAGE_ONLY, false]
      languages:
        en:
          name: English
        fr:
          name: Français (Beta)
          build: true
```

sitemap.xml

```xml
    <url>
         <loc>https://development.docs.coregames.com/api/coremath/</loc>
         <lastmod>2022-02-07</lastmod>
         <changefreq>daily</changefreq>
         <xhtml:link rel="alternate" hreflang="en" href="https://development.docs.coregames.com/en/api/coremath/"/>
         <xhtml:link rel="alternate" hreflang="fr" href="https://development.docs.coregames.com/fr/api/coremath/"/>
         
    </url>
```

**Comment by ultrabug** [comment]
It's not wrong to point to its own language, Google actually recommends it

**Comment by Stanzilla** [comment]
> It's not wrong to point to its own language, Google actually recommends it

Yeah but in this case it points to /en/ which is not being built

**Comment by ultrabug** [comment]
Oh right... I'll figure something out indeed

**Comment by Stanzilla** [comment]
Actually, I just noticed that with this config, it actually builds `/en/` because `build` defaults to `true` but yeah, the sitemap still does the above.

**Comment by ultrabug** [comment]
Alright, 0.42 with (all) the fixes, thanks

**Comment by Stanzilla** [comment]
> Alright, 0.42 with (all) the fixes, thanks

Can confirm fixed, thanks as always!

Has there been any movement on maybe integrating this into mkdocs itself? I think it would benefit greatly from it.

**Comment by Stanzilla** [comment]
Oh I meant the whole plugin

**Pull request merged**